packages/apollo-collaboration-server/src/counters/counters.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Counter, CounterDocument } from 'apollo-schemas'
+import { Model } from 'mongoose'
+
+@Injectable()
+export class CountersService {
+  constructor(
+    @InjectModel(Counter.name)
+    private readonly counterModel: Model<CounterDocument>,
+  ) {}
+
+  private readonly logger = new Logger(CountersService.name)
+
+  async getNextSequenceValue(sequenceName: string): Promise<number> {
+    const sequenceDocument = await this.counterModel
+      .findOneAndUpdate(
+        { id: sequenceName },
+        { $inc: { sequenceValue: 1 } },
+        { new: true, upsert: true },
```

Comment:
I had to add "upsert" here, which inserts a new document if there isn't one yet, to make it work on a fresh database.